algorithms/src/merkle_tree/merkle_tree.rs

```diff
@@ -322,10 +317,10 @@ impl<P: MerkleParameters + Send + Sync> MerkleTree<P> {
 
         if path.len() != Self::DEPTH as usize {
             let empty_hash = self.parameters.hash_empty()?;
-            path.push((self.tree[0].clone(), empty_hash));
+            path.push(empty_hash);
 
-            for &(ref hash, ref sibling_hash) in &self.padding_tree {
-                path.push((hash.clone(), sibling_hash.clone()));
+            for &(ref _hash, ref sibling_hash) in &self.padding_tree {
+                path.push(sibling_hash.clone());
```

Comment:
If I'm not mistaken, the `padding_tree` is only ever populated with `(current_hash.clone(), empty_hash.clone())` tuples; doesn't this mean it is now redundant? Alternatively, doesn't it mean that the `sibling_hash` is always equal to `self.parameters.hash_empty()`?

Reply:
Based on this implementation, I think you're right.

Although given the padding tree is already computed by the time this function is called, I suppose it isn't significantly affecting performance.

I do think this padding approach is opinionated in its instantiation, though likely adequate for us.